### PR TITLE
fix subpackage check for setup.py generation

### PR DESCRIPTION
**Pull request by abn**
This is a forward port of python-poetry/poetry#1626.

Resolves: python-poetry/poetry#2283

### CODE REVIEW

**Comment by abn** [comment]
Unrelated failures in CI.

File: poetry/core/masonry/builders/sdist.py [hunk 1]
```diff
@@ -240,7 +240,15 @@ def find_nearest_pkg(rel_path):
             if from_top_level == ".":
                 continue
 
-            is_subpkg = any([filename.endswith(".py") for filename in filenames])
+            is_subpkg = any(
+                [filename.endswith(".py") for filename in filenames]
+            ) and not all(
+                [
```
Comment (stephsamson) [inline poetry/core/masonry/builders/sdist.py:246]: Could this be extracted out to a different variable to make it easier to read? It took me a while to understand that this was checking that all of the Python files are not part of the excluded files.

Comment (abn) [reply poetry/core/masonry/builders/sdist.py:246]: I agree; but I wanted to keep this at par with the fix we have already merged in poetry pr. That might be a seperate change.

Comment (stephsamson) [reply poetry/core/masonry/builders/sdist.py:246]: Okay, makes sense. Then this PR looks good to go.

**Pull request merged**